### ISSUE TITLE
Autocompletion of UIKit

### ISSUE THREAD

**Issue by LinusU** [vknabel/vscode-swift-development-environment#8]
I'm not 100% sure that it's specifically UIKit that is the problem, but I'm having problems with autocomplete not giving any relevant suggestions e.g. when writing `self.` inside a class that inherits from e.g. `UIView`.

Unfortunately, this is the majority of my code so I'm currently coding without proper autocompletion 😱

I'd be happy to help in any way I can, be that more investigation, or getting my hands dirty with the code 👍

**Comment by vknabel** [comment]
@LinusU actually this affects all third party APIs. Only your own code will be autocompleted.

I will have a deeper look on the current state of possibilities and report it in here.

**Comment by vknabel** [comment]
@LinusU within the latest commit, I could make autocompletion for `AppKit` work. If you can make compilation of a single swift file which uses `UIKit` work, we could theoretically support it in here. The problems are the different target and SDK. 

Then we would need to slightly refactor `loadArgsImportPaths` to be configurable and we could ship it. (Maybe I will release the AppKit and fixes release and thereafter the UIKit release)

**Comment by LinusU** [comment]
Awesome 👍 

I'm currently not using `swift build` directly, but are using `xcodebuild` together with a generated Xcode-project. Here is the setup: https://github.com/LinusU/Soon

You can make it print the `swiftc` invocations, so I might be able to narrow everything down to a single call to compile one small file.

**Comment by LinusU** [comment]
I've managed to narrow it down to this quite simple invocation:

```sh
swiftc -sdk /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk -target arm64-apple-ios11.0 -swift-version 4 main.swift
```

This is my `main.swift`, just to test:

```swift
import UIKit

class TestButton: UIButton {
    var testString = "Test"
}
```

It seems like `-sdk` can be specified multiple times, so maybe it's just a matter of adding that path as well? 🙌

**Comment by vknabel** [comment]
@LinusU Hi I got working autocompletion for UIKit 🎉

In order to get autocompletion for UIKit, you need to add `"sde.sourcekit.compilerOptions": ["-target", "arm64-apple-ios11.0"]` to the workspace settings and restart your vscode window.

Could you try it out before I publish it?
[vscode-swift-development-environment-2.2.0.vsix.zip](https://github.com/vknabel/vscode-swift-development-environment/files/1808483/vscode-swift-development-environment-2.2.0.vsix.zip)

**Comment by LinusU** [comment]
Awesome! It works!!

Just found one small issue, that is probably not related though: https://github.com/vknabel/vscode-swift-development-environment/issues/9

**Comment by vknabel** [comment]
Great to hear! I just released the 2.2.0 update including this change!
Thank you for being such a great help.

**Comment by LinusU** [comment]
No problem at all, thank you for the incredible work 🚀
